### PR TITLE
[algo] fix: correct seq-mean loss denominator for DP-invariant aggregation

### PR DESCRIPTION
**Pull request by dubin555**
### What does this PR do?

Fixes the `seq-mean` loss denominator in `agg_loss()` which causes the effective loss to be `dp_size²` times too small in multi-GPU training.

**Background:** PR #5366 correctly fixed `seq-mean-token-sum-norm` by adding `* dp_size` compensation for FSDP gradient averaging. However, the shared denominator used by `seq-mean-token-mean` and `seq-mean-token-sum` was not fixed — it still multiplies `global_batch_size * dp_size` instead of dividing.

**Bug (line 816):**
```python
# FSDP divides gradients by dp_size, so loss must compensate by * dp_size.
# token-mean (correct):  loss = local_sum / global_tokens * dp_size
# seq-mean (BUGGY):      seq_denominator = global_batch_size * dp_size  ← wrong direction!
```

**Proof:** With dp_size=4, global_batch_size=32, equal per-token losses:
- token-mean: `L_eff = 1.0` ✓
- seq-mean-token-mean (buggy): `L_eff = 0.0625` (16× too small = dp_size²)
- seq-mean-token-mean (fixed): `L_eff = 1.0` ✓

**Affected algorithms:**
- **GSPO** — hardcodes `loss_agg_mode="seq-mean-token-mean"`
- **SAPO** — hardcodes `loss_agg_mode="seq-mean-token-mean"`
- **REINFORCE** — defaults to `loss_agg_mode="seq-mean-token-sum"`
- Any user config with `seq-mean-*` modes

**Trigger conditions:** new engine path (`use_legacy_worker_impl=disable`) + `dp_size > 1`

**Fix:** One-line change:
```python
# Before:
seq_denominator = global_batch_size * dp_size
# After:
seq_denominator = global_batch_size / dp_size
```

This makes seq-mean consistent with token-mean: `loss = local_seq_sum / (global_batch_size / dp_size) = local_seq_sum * dp_size / global_batch_size`.

### Checklist Before Starting

- [x] Search for similar PRs: https://github.com/verl-project/verl/pulls?q=is%3Apr+agg_loss+seq-mean+dp
- [x] Format the PR title as `[module] type: description`
- [x] Related: PR #5366 fixed seq-mean-token-sum-norm but missed other seq-mean modes

### Tests

7 unit tests (5 fail without fix, all pass with fix):
```bash
python3 -m pytest tests/unit/test_agg_loss_dp_scaling.py -v  # 7 passed
```

Tests cover all seq-mean modes, DP invariance property, uneven sequence lengths, and legacy path compatibility.

### CODE REVIEW

**Comment by wuxibin89** [comment]
@dubin555 You seem not using the latest version?
